banking-bench/src/main.rs

```diff
@@ -370,7 +398,7 @@ fn main() {
                     bank.transaction_count(),
                     txs_processed
                 );
-                assert!(txs_processed < bank.transaction_count());
+                //assert!(txs_processed < bank.transaction_count());
```

Comment:
This didn't seem safe in the following scenario: 

1. We create a new bank and add it to poh recorder: https://github.com/solana-labs/solana/blob/fc4e4b229c7d269f150b1174ce3127438827fbe9/banking-bench/src/main.rs#L418-L428
2. Before we can send the transactions for the next iteration, this bank is removed from poh recorder
3. We send the transactions for the next iteration, none of them land
4. `check_txs` returns because there's no bank: https://github.com/solana-labs/solana/blob/fc4e4b229c7d269f150b1174ce3127438827fbe9/banking-bench/src/main.rs#L57
5. No transactions were landed in this bank so `txs_processed == bank.transaction_count()`

Reply:
I am almost sure I never see this assert fail for described scenario. I mean why would #2 happen?

Reply:
I think it can happen if all the work after `set_bank()` takes longer than 350ms, things like the blockhash refresh, signature clearing, etc.